dali/operators/reader/loader/file_label_loader.cc

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2017-2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dirent.h>
+#include <errno.h>
+#include <memory>
+
+#include "dali/core/common.h"
+#include "dali/operators/reader/loader/file_label_loader.h"
+#include "dali/util/file.h"
+#include "dali/operators/reader/loader/utils.h"
+
+namespace dali {
+
+
+inline void assemble_file_list(const std::string& path, const std::string& curr_entry, int label,
+                        std::vector<std::pair<std::string, int>> *file_label_pairs) {
+  std::string curr_dir_path = path + "/" + curr_entry;
+  DIR *dir = opendir(curr_dir_path.c_str());
+
+  struct dirent *entry;
```

Comment:
```suggestion
  dirent *entry;
```
we're writing in C++ ;)

Reply:
fixed